glean-core/rlb/src/private/labeled.rs

```diff
@@ -0,0 +1,389 @@
+    ///
+    /// # Arguments
+    ///
+    /// * `error` - The type of error
+    /// * `ping_name` - represents the optional name of the ping to retrieve the
+    ///   metric for. Defaults to the first value in `send_in_pings`.
+    ///
+    /// # Returns
+    ///
+    /// The number of errors reported.
+    fn test_get_num_recorded_errors<'a, S: Into<Option<&'a str>>>(
+        &self,
+        error: ErrorType,
+        ping_name: S,
+    ) -> i32 {
+        dispatcher::block_on_queue();
+
+        crate::with_glean_mut(|glean| {
+            glean_core::test_get_num_recorded_errors(
+                &glean,
+                self.0.get_submetric().meta(),
+                error,
+                ping_name.into(),
+            )
+            .unwrap_or(0)
+        })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::ErrorType;
+    use crate::common_test::{lock_test, new_glean};
+    use crate::destroy_glean;
+    use crate::private::{BooleanMetric, CounterMetric, LabeledMetric, StringMetric};
+    use crate::CommonMetricData;
+
+    #[test]
+    fn test_labeled_counter_type() {
+        let _lock = lock_test();
+
+        let _ = new_glean(None, true);
+
+        let metric: LabeledMetric<CounterMetric> = LabeledMetric::new(
+            CommonMetricData {
+                name: "labeled_counter".into(),
+                category: "labeled".into(),
+                send_in_pings: vec!["test1".into()],
+                disabled: false,
+                ..Default::default()
+            },
+            None,
+        );
+
+        metric.get("label1").add(1);
+        metric.get("label2").add(2);
+        assert_eq!(1, metric.get("label1").test_get_value("test1").unwrap());
+        assert_eq!(2, metric.get("label2").test_get_value("test1").unwrap());
+    }
+
+    #[test]
+    fn test_other_label_with_predefined_labels() {
+        let _lock = lock_test();
+
+        let _ = new_glean(None, true);
+
+        let metric: LabeledMetric<CounterMetric> = LabeledMetric::new(
+            CommonMetricData {
+                name: "labeled_counter".into(),
+                category: "labeled".into(),
+                send_in_pings: vec!["test1".into()],
+                disabled: false,
+                ..Default::default()
+            },
+            Some(vec!["foo".into(), "bar".into(), "baz".into()]),
+        );
+
+        metric.get("foo").add(1);
+        metric.get("foo").add(2);
+        metric.get("bar").add(1);
+        metric.get("not_there").add(1);
+        metric.get("also_not_there").add(1);
+        metric.get("not_me").add(1);
+
+        assert_eq!(3, metric.get("foo").test_get_value(None).unwrap());
+        assert_eq!(1, metric.get("bar").test_get_value(None).unwrap());
+        assert!(metric.get("baz").test_get_value(None).is_none());
+        // The rest all lands in the __other__ bucket.
+        assert_eq!(3, metric.get("not_there").test_get_value(None).unwrap());
```

Comment:
Comment says `__other__` while the code tries to get `not_there`.
We either should ask for `__other__`, or call out that this maps to `__other__` anyway